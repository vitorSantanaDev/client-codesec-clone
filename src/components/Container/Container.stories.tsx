import { ComponentStory, ComponentMeta } from '@storybook/react'
import Container from '.'

export default {
  title: 'Container',
  component: Container,
  args: {
    children: <h1>Nex app</h1>
  }
} as ComponentMeta<typeof Container>

export const Default: ComponentStory<typeof Container> = (args) => (
  <Container {...args} />
)
