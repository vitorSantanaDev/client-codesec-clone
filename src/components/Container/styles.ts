import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    margin: 0 auto;
    max-width: 1200px;
    padding: ${theme.spacings.large};

    display: flex;
    justify-content: center;
  `}
`
