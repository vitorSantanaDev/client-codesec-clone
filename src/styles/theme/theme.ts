export const theme = {
  colors: {
    redPrimary: '#E83752',
    redLight: '#FA5770',
    redDark: '#D4203B',
    redError: '#EB3D3D',
    redGradient: 'linear-gradient(243.46deg, #FA5770 16.66%, #E83752 83.34%)',
    redGradientDark:
      'linear-gradient(225.05deg, #FA5770 0%, rgba(232, 55, 82, 0) 100%)',
    redGradientLight:
      'linear-gradient(225.05deg, rgba(250, 87, 112, 0.1) 0%, rgba(232, 55, 82, 0) 50%)',
    greenSucsses: '#32D957',
    white: '#FFFFFF',
    whiteLight: '#F5F0F1',
    whiteDark: '#EDE4E5',
    black: '#121111',
    blackLight: '#1C1B1B',
    blackDark: '#0A0909',
    gray1: '#141414',
    gray2: '#2C2C2C',
    gray3: '#444444',
    gray4: '#5C5C5C',
    gray5: '#747474',
    gray6: '#8C8C8C',
    gray7: '#A3A3A3'
  },
  font: {
    family: {
      default: "'Raleway', sans-serif"
    },
    sizes: {
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
      xxhuge: '8rem'
    }
  },
  media: {
    lteMedium: '(max-width: 768px)',
    lteLarge: '(max-width: 820px)',
    ltexLarge: '(max-width: 912px)',
    lteSmall: '(max-width: 420px)',
    lteXSmall: '(max-width: 280px)'
  },
  spacings: {
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
    xxhuge: '8rem'
  }
}
