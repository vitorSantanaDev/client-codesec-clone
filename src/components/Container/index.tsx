import { ReactNode } from 'react'
import * as S from './styles'

type ContainerProps = {
  children: ReactNode
}

const Container = ({ children }: ContainerProps) => (
  <S.Wrapper>{children}</S.Wrapper>
)

export default Container
