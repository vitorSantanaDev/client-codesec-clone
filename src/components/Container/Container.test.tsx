import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/theme/render-theme'
import Container from '.'

describe('<Container />', () => {
  it('should render the heading', () => {
    const { container } = renderTheme(
      <Container>
        <h1>Next App</h1>
      </Container>
    )
    const heading = screen.getByRole('heading', { name: /next app/i })
    expect(heading).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
