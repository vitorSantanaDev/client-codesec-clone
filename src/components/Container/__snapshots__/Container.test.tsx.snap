// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render the heading 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  max-width: 1200px;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
>
  <h1>
    Next App
  </h1>
</div>
`;
